### PR TITLE
Nodebb oidc blog post

### PR DESCRIPTION
**Pull request by mooreds**

### CODE REVIEW

File: site/_posts/2020-05-04-setting-up-single-sign-on-for-nodebb.md [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+
+<!--more-->
+
+## The dream
+
+Imagine you build out an application. Let's say you are building the world's best todo tracking app. Everyone loves it and it becomes wildly popular. Users are adding tasks and checking them off with abandon. Your customers clamor for a place to share all their task management tips. 
+
+Enter nodebb. You download and install it. The forum works great! Then you notice that every one of your users now must sign up for another username and password. Boo. 
+
+Luckily, you planned ahead and used an identity management system with OpenID Connect (OIDC) support. You realize you can set up nodebb to use this existing system to provide SSO to your users. All your todo app users will be able to log in and post in your forum with the same username and password they use for your app. Ah, the glory of single sign on.
```
Comment (pgeoly) [inline site/_posts/2020-05-04-setting-up-single-sign-on-for-nodebb.md:23]: I see sign-on hyphenated by most people and companies in the auth space. You're call but I vote for hyphenating it.
```suggestion
Ah, the glory of single sign-on. 
```

File: site/_posts/2020-05-04-setting-up-single-sign-on-for-nodebb.md [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+
+You'll also need an OIDC server. In this post, I'll be using FusionAuth as my identity provider ([here's the five minute setup guide](https://fusionauth.io/docs/v1/tech/5-minute-setup-guide)), but the nodebb configuration will be much the same with any OIDC server. FusionAuth will live at `http://local.fusionauth.io`.
+
+(Wait, what if you aren't using an OIDC compliant identity management system as your SSO server? Well, FusionAuth is free for unlimited users and here's [migration instructions](https://fusionauth.io/docs/v1/tech/tutorials/migrate-users). Get going!)
+
+## Setting up the SSO server
+
+First, create an application in FusionAuth. I'm going to call mine, creatively, "nodebb". Note that you can create as many applications as you want. If you were really building out "The Dream", you'd also have a 'todo app' FusionAuth application, but for the purposes of this tutorial we'll only set up the nodebb FusionAuth config. Users will have the same username and password for all applications, but may not be able to access every one of them. (Access is controlled by user registrations, which we'll cover below.)
```
Comment (pgeoly) [inline site/_posts/2020-05-04-setting-up-single-sign-on-for-nodebb.md:37]: Tautology - delete "the purposes of"
```suggestion
If you were really building out "The Dream", you'd also have a 'todo app' FusionAuth application, but for this tutorial, we'll only set up the nodebb FusionAuth config. Users will have the same username and password for all applications, but may not be able to access every one of them. (Access is controlled by user registrations, which we'll cover below.)
```

File: site/_posts/2020-05-04-setting-up-single-sign-on-for-nodebb.md [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+
+This is a better user experience, so turning off local login is what I'd recommend. Note that turning off local login and registration doesn't mean that users can't modify their profile. They absolutely can, as you can see from this nodebb profile edit screen for my example user:
+
+{% include _image.liquid src="/assets/img/blogs/nodebb-single-sign-on/editprofile.png" alt="A user editing their profile." class="img-fluid" figure=false %}
+
+Essentially, customer username and password information are kept in the identity server, and almost everything else is stored in nodebb.
+
+## FusionAuth features
+
+The steps above should work with any compliant SSO identity provider. Now we're going to cover features that aren't entirely part of the OIDC standard, but are useful for managing users. If you are using a different identity management server, there should be analogs for the features outlined below.
```
Comment (pgeoly) [inline site/_posts/2020-05-04-setting-up-single-sign-on-for-nodebb.md:108]:
```suggestion
The steps above should work with any compliant SSO identity provider. Now we're going to cover features that aren't entirely part of the OIDC standard but are useful for managing users. If you are using a different identity management server, there should be analogs for the features outlined below.
```

File: site/_posts/2020-05-04-setting-up-single-sign-on-for-nodebb.md [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+
+### User management
+
+If you lock the user account in the SSO server, the user won't be able to login. This is helpful when managing a forum; if a user is not being a good community member you can disable their account either temporarily or permanently. In FusionAuth, you'd navigate to the users list and then click on the "Lock" icon (you'll be warned that a locked user can't login):
```
Comment (pgeoly) [inline site/_posts/2020-05-04-setting-up-single-sign-on-for-nodebb.md:112]: Use the verb log in. Add apostrophe to users'
```suggestion
If you lock the user account in the SSO server, the user won't be able to log in. This is helpful when managing a forum; if a user is not being a good community member you can disable their account either temporarily or permanently. In FusionAuth, you'd navigate to the users' list and then click on the "Lock" icon (you'll be warned that a locked user can't log in):
```

File: site/_posts/2020-05-04-setting-up-single-sign-on-for-nodebb.md [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+
+{% include _image.liquid src="/assets/img/blogs/nodebb-single-sign-on/lock-user-from-list.png" alt="The user list screen where you can lock a user account." class="img-fluid" figure=false %}
+
+And here's what the locked out user would see when they tried to log in:
+
+{% include _image.liquid src="/assets/img/blogs/nodebb-single-sign-on/lockedaccount.png" alt="The login screen for a user whose account has been locked." class="img-fluid" figure=false %}
+
+Of course, people can change. So an admin can unlock an account anytime, either using the FusionAuth admin console, or with the [FusionAuth API](https://fusionauth.io/docs/v1/tech/apis/users#reactivate-a-user).
```
Comment (pgeoly) [inline site/_posts/2020-05-04-setting-up-single-sign-on-for-nodebb.md:120]: Delete comma after "console"
```suggestion
Of course, people can change. So an admin can unlock an account anytime, either using the FusionAuth admin console or with the [FusionAuth API](https://fusionauth.io/docs/v1/tech/apis/users#reactivate-a-user).
```

File: site/_posts/2020-05-04-setting-up-single-sign-on-for-nodebb.md [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+
+Next, we'll associate the user you previously created with the admin role. We do this on the user edit page. This role will be passed to nodebb. That will allow this user access to the nodebb admin screens. When you are done, the role should be shown on the user's page:
+
+{% include _image.liquid src="/assets/img/blogs/nodebb-single-sign-on/user-with-admin-role.png" alt="The user is associated with the admin role." class="img-fluid" figure=false %}
+
+We also need to update the plugin with the correct "Roles claim". Sign in as an admin to nodebb. (If you disabled local login already, you may need to use a special login path; you can always go directly to the local login screen by adding `/login?local=1` to your nodebb base URL: `http://localhost:4567/login?local=1`.)
+
+Update the plugin configuration; set the "Roles claim" be `roles` and restart nodebb. Now you can log in to nodebb with this user, or any other with the admin role in FusionAuth, and view forum admin pages. Admin users will be able to move topics, edit posts and perform all the other tasks needed to keep your todo app forum users happy.
```
Comment (pgeoly) [inline site/_posts/2020-05-04-setting-up-single-sign-on-for-nodebb.md:136]: Delete comma after FusionAuth
```suggestion
Update the plugin configuration; set the "Roles claim" be `roles` and restart nodebb. Now you can log in to nodebb with this user, or any other with the admin role in FusionAuth and view forum admin pages. Admin users will be able to move topics, edit posts and perform all the other tasks needed to keep your todo app forum users happy.
```

File: site/_posts/2020-05-04-setting-up-single-sign-on-for-nodebb.md [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+
+After this change, users will see a registration prompt on the login page: "Don't have an account? Create an account". When they click on it, they'll be prompted to register, with a screen like below:
+
+{% include _image.liquid src="/assets/img/blogs/nodebb-single-sign-on/user-registration.png" alt="The registration screen when self registration has been enabled." class="img-fluid" figure=false %}
+
+After a user creates an account, they'll be logged in. At this point, they'd only have the ability to log in to nodebb.
+
+### Theming the pages
+
+You may have noticed that the login and logout pages didn't look like the nodebb pages. FusionAuth gives you the ability to theme your login, logout and other authentication pages. This involves changing templates which you can access via the admin screen.
```
Comment (pgeoly) [inline site/_posts/2020-05-04-setting-up-single-sign-on-for-nodebb.md:154]: Add comma to list.
```suggestion
You may have noticed that the login and logout pages didn't look like the nodebb pages. FusionAuth gives you the ability to theme your login, logout, and other authentication pages.
```

File: site/_posts/2020-05-04-setting-up-single-sign-on-for-nodebb.md [hunk 8]
```diff
@@ -0,0 +1,163 @@
...
+
+### Theming the pages
+
+You may have noticed that the login and logout pages didn't look like the nodebb pages. FusionAuth gives you the ability to theme your login, logout and other authentication pages. This involves changing templates which you can access via the admin screen. 
+
+The [FusionAuth theme documentation](https://fusionauth.io/docs/v1/tech/themes/) outlines the templates and gives an example of customizing the login page.
+
+## Conclusion
+
+nodebb is a powerful piece of forum software. Standing up a forum lets your users communicate with each other and can be a powerful mechanism for knowledge sharing, search engine optimization and community building.
```
Comment (pgeoly) [inline site/_posts/2020-05-04-setting-up-single-sign-on-for-nodebb.md:160]: Add comma to list.
```suggestion
nodebb is a powerful piece of forum software. Standing up a forum lets your users communicate with each other and can be a powerful mechanism for knowledge sharing, search engine optimization, and community building.
```

File: site/_posts/2020-05-04-setting-up-single-sign-on-for-nodebb.md [hunk 9]
```diff
@@ -0,0 +1,163 @@
...
+
+Using an SSO server such as FusionAuth lets your users minimize the number of usernames and passwords they must manage. It also allows you to control access to your applications from one screen, also known as "The Dream". Using a single sign on server for centralized user management will make both your life and your users' lives easier.
```
Comment (pgeoly) [inline site/_posts/2020-05-04-setting-up-single-sign-on-for-nodebb.md:162]: Hyphenate sign-on.
```suggestion
Using a single sign-on server for centralized user management will make both your life and your users' lives easier.
```

File: site/_posts/2020-05-04-setting-up-single-sign-on-for-nodebb.md [hunk 10]
```diff
@@ -0,0 +1,163 @@
+---
+layout: blog-post
+title: Setting up single sign on for nodebb
```
Comment (pgeoly) [inline site/_posts/2020-05-04-setting-up-single-sign-on-for-nodebb.md:3]:
```suggestion
title: Setting up single sign-on for nodebb 
```

**Comment by mooreds** [comment]
Also review image widths.

**Pull request merged**